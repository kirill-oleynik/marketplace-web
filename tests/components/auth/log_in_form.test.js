const React = require('react');
const renderer = require('react-test-renderer');
const { LogInForm } = require('../../../app/components/auth/log_in_form.js');

describe('#render', () => {
  test('it renders correctly', () => {
    const tree = renderer.create(
      <LogInForm
        t={(translation) => translation}
        onSubmit={() => {}}
        errors={{}}
        revalidation={{
          form: {
            values: [{
              rememberMe: false
            }]
          },
          onChange: (value) => () => {}
        }}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
