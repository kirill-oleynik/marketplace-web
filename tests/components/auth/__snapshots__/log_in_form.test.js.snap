// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    className="main-input mb-15"
  >
    <div
      className="main-input__field-box"
    >
      <input
        className="main-input__field"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        className="main-input__label"
        htmlFor="email"
      >
        fields.email
      </label>
    </div>
  </div>
  <div
    className="main-input mb-20 with-icon"
  >
    <div
      className="main-input__field-box"
    >
      <i
        className="main-input__icon icon icon-eye"
        onClick={[Function]}
      />
      <input
        className="main-input__field"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <label
        className="main-input__label"
        htmlFor="password"
      >
        fields.password
      </label>
    </div>
  </div>
  <button
    className="main-btn main-btn--blue main-btn--lg disabled w-100 mb-20"
    disabled={false}
    type="submit"
  >
    submit
  </button>
  <div
    className="login-form__info-wrap mb-20"
  >
    <a
      className="login-form__link mb-20"
      href="/"
    >
      forgotPassword
    </a>
    <div>
      <label
        className="checkbox mb-20"
        htmlFor="checkbox-rememberMe"
      >
        <input
          checked={undefined}
          className="checkbox__input"
          id="checkbox-rememberMe"
          name="rememberMe"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkbox__icon icon"
        />
        <span
          className="checkbox__text"
        >
          rememberMe
        </span>
      </label>
    </div>
  </div>
</form>
`;
